src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -269,11 +269,13 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
         <_AndroidApplicationSharedLibraryPath>$(IntermediateOutputPath)app_shared_libraries\</_AndroidApplicationSharedLibraryPath>
 	<_ResolvedUserAssembliesHashFile>$(IntermediateOutputPath)resolvedassemblies.hash</_ResolvedUserAssembliesHashFile>
 
+	<AndroidDexTool   Condition=" '$(AndroidDexTool)' != 'dx' ">d8</AndroidDexTool>
```

Comment:
I moved this `$(AndroidDexTool)` reassignment a few lines up so that the new properties are evaluated _after_ this reassignment but _before_ `$(AndroidEnableProguard)` gets reassigned. That way they see `d8` as expected when it is turned on for an empty or invalid `$(AndroidDexTool)` from the app project, but they see the original `$(EnableProguard)` and `$(AndroidEnableProguard)` values from the app project.